src/structurizr/model/component.py

```diff
@@ -104,13 +105,15 @@ def __init__(
         self.code_elements = set(code_elements)
         self.size = size
 
+        self.tags.add(Tags.COMPONENT)
+
     @classmethod
-    def hydrate(cls, component_io: ComponentIO) -> "Component":
+    def hydrate(cls, component_io: ComponentIO, container: "Container") -> "Component":
         """"""
-        # TODO (midnighter): Initialization requires `parent`.
         return cls(
-            name=component_io.name,
-            description=component_io.description,
+            **super().hydrate_arguments(component_io),
```

Comment:
Did you try `self`? It should call the nearest method up the MRO. `super` should only be used inside `hydrate_arguments` if it is composing arguments from parent classes, too.

```suggestion
            **self.hydrate_arguments(component_io),
```